nrf_modem/doc/architecture.rst

```diff
@@ -3,19 +3,54 @@
 Library architecture
 ####################
 
-The purpose of the Modem library is to communicate with the nRF9160 modem firmware.
-This is done by sending events and getting responses over the Inter Processor Communication (IPC) peripheral.
-To facilitate data transfer between the application and the modem, payload data is set in the shared RAM.
+The Modem library is the application interface to the nRF9160 modem firmware, which contains a full IP and DTLS/TLS stack as well as GNSS.
+It provides the standard Socket APIs, to facilitate application development and additional APIs to manage the modem and perform full modem firmware updates.
 
-The modem contains a full IP and DTLS/TLS stack as well as GNSS.
-The library internally communicates over a Remote Procedure Call (RPC) protocol using the IPC and the shared RAM to utilize the protocol stacks inside the modem.
-You can dispatch data to and from the different modules in the modem firmware by providing the right combinations of address families, types, and protocols to the Modem library.
-
-The Modem library aims to provide a standard programming interface on top of the RPC protocol, to facilitate application development.
+The library implements a communication between the Application and Modem cores on the nRF9160 via the RPC protocol, using the Inter Processor Communication (IPC) peripheral and a shared region of RAM.
 
 The following figure shows a simplified Modem library architecture.
 
 .. figure:: images/nrf_modem_architecture.svg
    :alt: Modem library architecture diagram
 
    Modem library architecture diagram
+
+Shared memory configuration
+###########################
+
+The Modem library implements a communication between the application core and the modem core using a shared memory area to exchange data.
```

Comment:
" .... implements a communication **interface** between the .... " ?

Reply:
ok, I will fix that